test/functional/ui/inccommand_user_spec.lua

```diff
@@ -0,0 +1,329 @@
+local helpers = require('test.functional.helpers')(after_each)
+local Screen = require('test.functional.ui.screen')
+local clear = helpers.clear
+local exec_lua = helpers.exec_lua
+local insert = helpers.insert
+local feed = helpers.feed
+local command = helpers.command
+
+-- Implementation of a :Replace command that's used for testing user inccommand
+local SetupReplaceCmd = [[
```

Comment:
I guess we could try having a completely different type of `preview` function entirely. I made one that mimics `:substitute` because that's currently the standard for how command preview should behave.